pykokkos/interface/type_inference.py

```diff
@@ -0,0 +1,279 @@
+import inspect
+from dataclasses import dataclass
+from typing import  Callable, Dict, Optional, Tuple, Union, List, Any
+import pykokkos.kokkos_manager as km
+from .execution_policy import MDRangePolicy, TeamPolicy, TeamThreadRange, RangePolicy, ExecutionPolicy
+from .views import View, ViewType
+from .layout import Layout
+from .data_types import DataType, DataTypeClass
+
+@dataclass
+class HandledArgs:
+    """
+    Class for holding the arguments passed to parallel_* functions
+    """
+
+    name: Optional[str]
+    policy: ExecutionPolicy
+    workunit: Callable
+    view: Optional[ViewType]
+    initial_value: Union[int, float]
+
+
+@dataclass
+class UpdatedTypes:
+    """
+    Class for storing inferred type annotation information 
+    (Making Pykokkos more pythonic by automatically inferring types)
+    """
+
+    workunit: Callable
+    inferred_types: Dict[str, str] # type information stored as string: identifier -> type
+    is_arg: set[str]
```

Comment:
should this be boolean if starts with `is`

Reply:
Got rid of this altogether